evadb/executor/create_function_executor.py

```diff
@@ -117,6 +118,48 @@ def handle_ludwig_function(self):
             self.node.metadata,
         )
 
+    def handle_sklearn_function(self):
+        """Handle sklearn functions
+
+        Use Sklearn's regression to train models.
+        """
+        try_to_import_sklearn()
+        from sklearn.linear_model import LinearRegression
+
+        assert (
+            len(self.children) == 1
+        ), "Create sklearn function expects 1 child, finds {}.".format(
+            len(self.children)
+        )
+
+        aggregated_batch_list = []
+        child = self.children[0]
+        for batch in child.exec():
+            aggregated_batch_list.append(batch)
+        aggregated_batch = Batch.concat(aggregated_batch_list, copy=False)
+        aggregated_batch.drop_column_alias()
+
+        arg_map = {arg.key: arg.value for arg in self.node.metadata}
+        model = LinearRegression()
+        model.fit(X=aggregated_batch.frames, y=arg_map["predict"])
```

Comment:
@jineetd `arg_map["predict"]` returns the column name that user wants to predict, so in your test case, the value is `rental_price`. We can use `aggregated_batch.frames[arg_map["predict"]]` instead to get the values. Notice we also need to remove the `rental_price` column from `X`, you can achieve so by `aggregated_batch.frames.drop(arg_map["predict"], axis=1)` .